File: demo/src/main/java/com/foodorderingsystem/demo/service/UserService.java
package com.foodorderingsystem.demo.service;

import com.foodorderingsystem.demo.wrapper.UserWrapper;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Map;

public interface UserService {
    
    ResponseEntity<String> signup(Map<String, String> requestMap);

    public ResponseEntity<String> login(Map<String, String> requestMap);

    public ResponseEntity<List<UserWrapper>> getAllUsers();

    public ResponseEntity<String> update(Map<String, String> requestMap);

    public ResponseEntity<String> checkToken();

    public ResponseEntity<String> changePassword(Map<String, String> requestMap);

    public ResponseEntity<String> forgotPassword(Map<String, String> requestMap);
}

File: Frontend/src/app/services/user.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  url = environment.apiURL;
  jsonHeader = {
    headers: new HttpHeaders().set('Content-Type', 'application/json'),
  };

  constructor(private http: HttpClient) {}

  signup(data: any) {
    return this.http.post(`${this.url}/user/signup`, data, this.jsonHeader);
  }

  forgotPassword(data: any) {
    return this.http.post(
      `${this.url}/user/forgotPassword`,
      data,
      this.jsonHeader
    );
  }

  login(data: any) {
    return this.http.post(`${this.url}/user/login`, data, this.jsonHeader);
  }

  checkToken() {
    return this.http.get(`${this.url}/user/check-token`);
  }

  changePassword(data: any) {
    return this.http.post(
      `${this.url}/user/changePassword`,
      data,
      this.jsonHeader
    );
  }

  getAllUsers() {
    return this.http.get(`${this.url}/user/get`);
  }

  update(data: any) {
    return this.http.post(`${this.url}/user/update`, data, this.jsonHeader);
  }
}
